src/Views/SuccessView.vala

```diff
@@ -45,37 +46,34 @@ public class Sideload.SuccessView : AbstractView {
-                    appstore_name, "settings://applications/permissions", _("System Settings"), _("Applications")
-                );
-            } else if (file is FlatpakBundleFile) {
-                secondary_label.label = _("Open it any time from the Applications Menu. Permissions can be changed in <a href='%s'>%s → %s…</a>").printf (
-                    /// TRANSLATORS: "System Settings" is related to the title of https://github.com/elementary/switchboard, "Applications" is related to the title of https://github.com/elementary/switchboard-plug-applications
-                    "settings://applications/permissions", _("System Settings"), _("Applications")
-                );
-            }
+            secondary_label_string = _("Open it any time from the Applications Menu.");
+
         } else if (view_type == SuccessType.ALREADY_INSTALLED) {
             if (app_name != null) {
                 primary_label.label = _("“%s” is already installed").printf (app_name);
             } else {
                 primary_label.label = _("This app is already installed");
             }
 
-            if (file is FlatpakRefFile) {
-                secondary_label.label = _("No changes were made. Visit %s for app information, updates, and to uninstall. Permissions can be changed in <a href='%s'>%s → %s…</a>").printf (
-                    /// TRANSLATORS: "System Settings" is related to the title of https://github.com/elementary/switchboard, "Applications" is related to the title of https://github.com/elementary/switchboard-plug-applications
-                    appstore_name, "settings://applications/permissions", _("System Settings"), _("Applications")
-                );
-            } else if (file is FlatpakBundleFile) {
-                secondary_label.label = _("No changes were made. Permissions can be changed in <a href='%s'>%s → %s…</a>").printf (
-                    /// TRANSLATORS: "System Settings" is related to the title of https://github.com/elementary/switchboard, "Applications" is related to the title of https://github.com/elementary/switchboard-plug-applications
-                    "settings://applications/permissions", _("System Settings"), _("Applications")
-                );
-            }
+            secondary_label_string = _("No changes were made.");
+        }
+
+        if (file is FlatpakRefFile) {
+            secondary_label_string += " ";
+            secondary_label_string += _("Visit %s for app information, updates, and to uninstall.").printf (
+                appstore_name
+            );
         }
 
+        /// TRANSLATORS: "System Settings" is related to the title of https://github.com/elementary/switchboard, "Applications" is related to the title of https://github.com/elementary/switchboard-plug-applications
+        var settings_path = _("System Settings → Applications");
+        var link_markup = "<a href='%s'>%s…</a>".printf ("settings://applications/permissions", settings_path);
```

Comment:
@danrabbit like this?

```suggestion
        /// TRANSLATORS: Includes an ellipsis (…) in English to signify the action will be performed in a new window
        var link_markup = _("<a href='%s'>%s…</a>").printf ("settings://applications/permissions", settings_path);
```

Reply:
No because that makes all this markup translatable again. I think it needs to go in that other string. The "System Settings → applications" one

Reply:
oh duh